src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEmbeddedAttributeSymbol.cs

```diff
@@ -22,25 +34,22 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     /// 4) It has System.Runtime.CompilerServices.CompilerGeneratedAttribute
     /// 5) It has a parameter-less constructor
     /// </summary>
-    internal sealed class SynthesizedEmbeddedAttributeSymbol : NamedTypeSymbol
+    internal abstract class SynthesizedEmbeddedAttributeSymbolBase : NamedTypeSymbol
```

Comment:
>SynthesizedEmbeddedAttributeSymbolBase [](start = 28, length = 38)

Do we really need an intermediate type? Could we leave its name as is and inherit from it?

Reply:
We could, but I thought we try to have a hierarchy with abstract or sealed nodes, rather than instantiatable intermediate nodes.

Reply:
> but I thought we try to have a hierarchy with abstract or sealed nodes, rather than instantiatable intermediate nodes.

I am not aware of such goal

---
In reply to: [215658001](https://github.com/dotnet/roslyn/pull/29180#discussion_r215658001) [](ancestors = 215658001)